espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/ScrollToPage.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.espressoserver.lib.handlers
+
+import androidx.test.espresso.EspressoException
+import androidx.test.espresso.contrib.ViewPagerActions
+import io.appium.espressoserver.lib.handlers.exceptions.AppiumException
+import io.appium.espressoserver.lib.handlers.exceptions.InvalidArgumentException
+import io.appium.espressoserver.lib.model.Element
+import io.appium.espressoserver.lib.model.ScrollToPageParams
+
+class ScrollToPage : RequestHandler<ScrollToPageParams, Void?> {
+
+    @Throws(AppiumException::class)
+    override fun handle(params: ScrollToPageParams): Void? {
+        val viewInteraction = Element.getViewInteractionById(params.elementId)
+        try {
+            val smoothScroll = params.smoothScroll
+            params.scrollTo?.let {
+                val scrollAction = when (it) {
+                    ScrollToPageParams.ScrollTo.FIRST -> ViewPagerActions.scrollToFirst(smoothScroll)
+                    ScrollToPageParams.ScrollTo.LAST -> ViewPagerActions.scrollToLast(smoothScroll)
+                    ScrollToPageParams.ScrollTo.LEFT -> ViewPagerActions.scrollLeft(smoothScroll)
+                    ScrollToPageParams.ScrollTo.RIGHT -> ViewPagerActions.scrollRight(smoothScroll)
+                }
```

Comment:
de we need else?

Reply:
We don't, actually.

Kotlin `when` statements only compile if
1. They're "exhaustive" (ie: every pathway is handled). 
1. They have an `else` statement to cover the default path

In this case, it's exhaustive so the else statement will be omitted. If another `ScrollTo` param is added, compilation will fail and we'll have to add it.